pkg/controller/versions/strategy_v2_1.go

```diff
@@ -323,6 +323,9 @@ func (v *versionStrategyV2_1) Render(ctx context.Context, cr *common.ControllerR
 		log.V(5).Info(fmt.Sprintf("rendering values:\n%s", string(rawValues)))
 	}
 
+	// Update spec.Istio back with the previous content merged with global.yaml
+	spec.Istio = v1.NewHelmValues(values)
```

Comment:
We shouldn't be updating the spec.  We should be updating appliedValues or just using the values as it was before.

Reply:
Ignore what I said.  It looks like there were more issues than just the isComponentEnabled() invocation (which is correct with this change), with part of the code using the merged `values` and other parts using the user specified `spec.Istio`.  Also, I verified that `spec` is not the actual resource, but the results after applying profiles.